### PR TITLE
Add extra security-related settings based on manage.py check --deploy

### PR DESCRIPTION
**Pull request by stevejalim**
C- on HTTP Observatory right now but these changes will lift that nicely https://observatory.mozilla.org/analyze/dev.future.nonprod.webservices.mozgcp.net

### CODE REVIEW

File: birdbox/birdbox/settings/base.py [hunk 1]
```diff
@@ -395,17 +395,38 @@ def set_whitenoise_headers(headers, path, url):
...
 
+CSRF_COOKIE_SECURE = True
+CSRF_COOKIE_HTTPONLY = True
+SECURE_BROWSER_XSS_FILTER = True
+SECURE_CONTENT_TYPE_NOSNIFF = True
+SESSION_COOKIE_SECURE = True
+X_FRAME_OPTIONS = "DENY"
+
+# Set header Strict-Transport-Security header
+SECURE_HSTS_SECONDS = int(os.environ.get("SECURE_HSTS_SECONDS", 3600))
```
Comment (pmac) [inline birdbox/birdbox/settings/base.py:421]: Why not use config helper here?

File: birdbox/birdbox/settings/base.py [hunk 2]
```diff
@@ -395,17 +395,38 @@ def set_whitenoise_headers(headers, path, url):
...
 SECURE_REDIRECT_EXEMPT = [
     r"^healthz/$",
     r"^readiness/$",
 ]
 
+CSRF_COOKIE_SECURE = True
+CSRF_COOKIE_HTTPONLY = True
+SECURE_BROWSER_XSS_FILTER = True
+SECURE_CONTENT_TYPE_NOSNIFF = True
+SESSION_COOKIE_SECURE = True
```
Comment (pmac) [inline birdbox/birdbox/settings/base.py:417]: Will this mean sessions don't work for local dev?

File: birdbox/birdbox/settings/base.py [hunk 3]
```diff
@@ -395,17 +395,38 @@ def set_whitenoise_headers(headers, path, url):
...
+X_FRAME_OPTIONS = "DENY"
+
+# Set header Strict-Transport-Security header
+SECURE_HSTS_SECONDS = int(os.environ.get("SECURE_HSTS_SECONDS", 3600))
+# Start with an hour, move to a year once we're settled.
+# Set to 0 via env to disable BEFORE deployment (!)
+
+# We do NOT want to roll all subdomains into the same HSTS setting
+# > Only set this to True if you are certain that all subdomains of your domain should be served exclusively via SSL.
+SECURE_HSTS_INCLUDE_SUBDOMAINS = False
```
Comment (pmac) [inline birdbox/birdbox/settings/base.py:427]: ➕

File: birdbox/birdbox/settings/base.py (hunk 1 above)
Comment (stevejalim) [reply birdbox/birdbox/settings/base.py:421]: D'oh. Oversight when lifting from an existing project

File: birdbox/birdbox/settings/base.py (hunk 2 above)
Comment (stevejalim) [reply birdbox/birdbox/settings/base.py:417]: I thought the same, so I checked beforehand and local dev seems ok - I'm assuming Django checks for localhost/127.0.0.1 and handles it specially

**Pull request merged**